main/FatesHistoryInterfaceMod.F90

```diff
@@ -4863,6 +4671,317 @@ subroutine update_history_dyn2(this,nc,nsites,sites,bc_in)
     return
   end subroutine update_history_dyn2
 
+  ! =========================================================================================
+
+  subroutine update_history_dyn2_ageclass(this,nc,nsites,sites,bc_in)
+
+    ! Arguments
+    class(fates_history_interface_type)             :: this
+    integer                 , intent(in)            :: nc   ! clump index
+    integer                 , intent(in)            :: nsites
+    type(ed_site_type)      , intent(inout), target :: sites(nsites)
+    type(bc_in_type)        , intent(in)            :: bc_in(nsites)
+
+    type(fates_cohort_type), pointer :: ccohort
+    type(fates_patch_type),  pointer :: cpatch
+    integer :: s, ft, iagepft, i_agefuel, iscag, iscagpft, i_fuel, i_scls, io_si
+    integer :: iscag_anthrodist  ! what is the equivalent age class for
+                                 ! time-since-anthropogenic-disturbance of secondary forest
+    real(r8) :: mort
+    real(r8) :: sapw_m             ! Sapwood mass (elemental, c,n or p) [kg/plant]
+    real(r8) :: struct_m           ! Structural mass ""
+    real(r8) :: leaf_m             ! Leaf mass ""
+    real(r8) :: fnrt_m             ! Fineroot mass ""
+    real(r8) :: store_m            ! Storage mass ""
+    real(r8) :: alive_m            ! Alive biomass (sap+leaf+fineroot+repro+storage) ""
+    real(r8) :: total_m            ! Total vegetation mass
+    real(r8) :: repro_m            ! Total reproductive mass (on plant) ""
+    real(r8) :: patch_area_div_site_area     ! Weighting based on patch area relative to site area
+    real(r8) :: patch_canarea_div_site_area  ! Weighting based on patch canopy area relative to site area
+    real(r8) :: cohort_n_div_site_area       ! Weighting based on cohort density relative to site area
```

Comment:
These kinds of variables could also be used in the non-ageclass subroutines for consistency and clarity.